File: tutorials/testing/python/array2d.py
from bridges.bridges import *
from bridges.array import *
from bridges.array2d import *
import sys

def main():

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    #Bridges title, description 
    bridges.set_title("Two-Dimensional Array Example")
    bridges.set_description("Illustrates a 2D array with visual attributes")


    # for 2D array 6x6
    num_rows = 6
    num_cols = 6
    my_array = Array2D(rows = num_rows, cols = num_cols)

    #populate the array
    for row in range(num_rows):
        for col in  range(num_cols):
            my_array[row, col].label = "El " + str(row*num_cols + col)

    #color some of the elements
    my_array[0, 0].color = "red"
    my_array[0, 3].color = "green"
    my_array[5, 0].color = "blue"
    my_array[5, 3].color = "magenta"
    my_array[1, 1].color = "cyan"
    my_array[2, 2].color = "yellow"

    #set visualizer type
    bridges.set_data_structure(my_array)

    #visualize tha array
    bridges.visualize()

if __name__ == "__main__":
    main()

File: tutorials/testing/c++/array2d.cpp
#include <string>
#include "Bridges.h"
#include "Array2D.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
	// title, description
	bridges.setTitle("Two-Dimensional Array Example");
	bridges.setDescription("Illustrates a 2D array with visual attributes");

	// create, populate 2D array
	int rows = 6;
	int cols = 6;
	Array2D<int> arr(rows, cols);

	// load array with its 1D address
	for (int i = 0; i < rows; i++)
		for (int j = 0; j < cols; j++)
			arr[i][j] = i * rows + j;

	for (int i = 0; i < rows; i++)
		for (int j = 0; j < cols; j++)
			arr.getElement(i, j).setLabel("El " + std::to_string(arr[i][j]));

	// set visual attributes
	arr.getElement(0, 0).setColor("red");
	arr.getElement(0, 3).setColor("green");
	arr.getElement(5, 0).setColor("blue");
	arr.getElement(5, 3).setColor("magenta");
	arr.getElement(1, 1).setColor("cyan");
	arr.getElement(2, 2).setColor("yellow");

	// visualize
	bridges.setDataStructure(&arr);
	bridges.visualize();

	return 0;
}
